src/visualization/FuncArrowShape.h

```diff
@@ -47,12 +48,14 @@ class FuncArrowShape : public ArrowShape {
          * \param sizeConstant an integer which scales the arrow
          * \param col the color of the arrow
          */
-        FuncArrowShape(Edge Arr, int sizeConstant, sf::Color col, int ind);
+        FuncArrowShape(vector<DrawNode>& Nodes, int sizeConstant, sf::Color col, int start, int end, int ind2);
```

Comment:
why do you pass the whole vector of `DrawNode`s to a single `FuncArrowShape`? As far as I can tell, you only need the `Drawnode`s for the starting and end position of the arrow? In that case, why not pass these 2 positions? Why must FuncArrowShape know that something like `DrawNode `even exists?

Reply:
That is true, i could maybe change it to one explicit DrawNode.
But i thought it would be fine, because FuncArrowShapes are just for our application and i pass the indices of the DrawNode by argument , too.